src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -46,21 +53,128 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(roundEntity);
   }
-  
 
-  private String getAuthToken() {
-    String clientId = "1c29ff191b444611a6d9dbb4a354642f";
-    String clientSecret = "6b92b6eb8fe24d5e98178407d3099bd8";
-    SpotifyApi spotifyApi = new SpotifyApi.Builder().setClientId(clientId).setClientSecret(clientSecret).build();
-    ClientCredentialsRequest clientCredentialsRequest = spotifyApi.clientCredentials().build();
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String url = getParameter(request, "playlist-link", "");
+    setVideoID(url);
+    response.sendRedirect("/game.html");
+  }
 
+  /**
+   * @return the request parameter, or the default value if the parameter was not specified by the
+   *     client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /**
+   * Use url to connect to YouTube API and set video ID.
+   *
+   * @param url
+   */
+  public void setVideoID(String url) {
+    String playlistID = getIDFromURL(url);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
     try {
-      ClientCredentials clientCredentials = clientCredentialsRequest.execute();
-      return clientCredentials.getAccessToken();
-    } catch (IOException | SpotifyWebApiException | ParseException e) {
-      System.out.println("Error: " + e.getMessage());
+      playlistItem = getPlaylistInfo(playlistID);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideos = parsePlaylistItem(playlistItemJson);
+    PLAYLIST_SIZE = playlistVideos.size();
+    videoID = getRandomVideo(playlistVideos);
+  }
+
+  /**
+   * Parse Youtube URL for playlist ID.
+   *
+   * @param url
+   * @return playlist ID
+   */
+  public String getIDFromURL(String url) {
```

Comment:
this is the kinda thing that could be tested. perhaps, we can add a todo for you to add tests later

Reply:
also method could be private

Reply:
+1 about testing in a follow-up PR. You can have similar tests to what you had for the getIdFromURL() function in JS, except they would be Java tests.